app/process/watcher/evm/watcher.go

```diff
@@ -17,47 +17,146 @@
 package evm
 
 import (
+	"context"
+	"errors"
 	"fmt"
+	"github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/hashgraph/hedera-sdk-go/v2"
 	"github.com/limechain/hedera-eth-bridge-validator/app/clients/evm/contracts/router"
 	"github.com/limechain/hedera-eth-bridge-validator/app/core/queue"
 	"github.com/limechain/hedera-eth-bridge-validator/app/domain/client"
 	qi "github.com/limechain/hedera-eth-bridge-validator/app/domain/queue"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/repository"
 	"github.com/limechain/hedera-eth-bridge-validator/app/domain/service"
+	"github.com/limechain/hedera-eth-bridge-validator/app/helper/timestamp"
 	"github.com/limechain/hedera-eth-bridge-validator/app/model/transfer"
 	c "github.com/limechain/hedera-eth-bridge-validator/config"
 	"github.com/limechain/hedera-eth-bridge-validator/constants"
 	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
 	"math/big"
 	"strconv"
 )
 
 type Watcher struct {
-	contracts service.Contracts
-	evmClient client.EVM
-	logger    *log.Entry
-	mappings  c.Assets
-	validator bool
+	repository  repository.Status
+	contracts   service.Contracts
+	evmClient   client.EVM
+	logger      *log.Entry
+	mappings    c.Assets
+	targetBlock uint64
+	validator   bool
 }
 
-func NewWatcher(contracts service.Contracts, evmClient client.EVM, mappings c.Assets, validator bool) *Watcher {
+func NewWatcher(
+	repository repository.Status,
+	contracts service.Contracts,
+	evmClient client.EVM,
+	mappings c.Assets,
+	startBlock int64,
+	validator bool) *Watcher {
+	targetBlock, err := evmClient.GetClient().BlockNumber(context.Background())
+	if err != nil {
+		log.Fatalf("Could not retrieve latest block. Error: [%s].", err)
+	}
+
+	if startBlock == 0 {
+		_, err := repository.GetLastFetchedTimestamp(contracts.Address().String())
```

Comment:
I think that it is misleading to store the blocknumber at the column for timestamp in the db. Wouldn't it be better to introduce another column for the block number?